samples/client/petstore/python/petstore_api/models/additional_properties_class.py

```diff
@@ -148,3 +148,7 @@ def __ne__(self, other):
         Returns true if both objects are not equal
         """
         return not self == other
+
+    def __hash__(self):
+      return hash((self.map_property,
+          self.map_of_map_property))
```

Comment:
Well, you will get `TypeError: unhashable type: 'dict'` here... I am not sure if this hash implementation make any sense since `dict` behaves the same way:

```
a = {1: 1}
b = {1: 1}
assert a == b  # OK
assert {a} == {b}  # AssertionError
```

Reply:
My implementation is based on https://docs.python.org/3/reference/datamodel.html
```
def __hash__(self):
    return hash((self.name, self.nick, self.color))
```
https://stackoverflow.com/questions/2909106/whats-a-correct-and-good-way-to-implement-hash suggests something similar.

But feel free to suggest other ways to implement `__hash__`

Reply:
Here the model is mutable. it would be wrong to implement a `__hash__` for it. [documentation](https://docs.python.org/3.5/reference/datamodel.html#object.__hash__) says:

> If a class defines mutable objects and implements an __eq__() method, it should not implement __hash__(), since the implementation of hashable collections requires that a key’s hash value is immutable (if the object’s hash value changes, it will be in the wrong hash bucket).

Reply:
Thanks for the explanation. Closing this :)